File: P12900.py
def solution(n):
    arr = [0] * 60001
    arr[1], arr[2] = 1, 2
    for i in range(3, n+1):
        arr[i] = (arr[i-1] + arr[i-2]) % 1000000007
    return arr[n]

File: P12900.java
public class P12900 {
    public int solution(int n) {
        int N = 1000000007;
        int[] arr = new int[60001];
        arr[1] = 1;
        arr[2] = 2;
        for (int i=3; i<=n; i++) arr[i] = (arr[i-1] + arr[i-2]) % N;
        return arr[n];
    }
}
